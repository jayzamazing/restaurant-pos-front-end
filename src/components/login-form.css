.login-form {
  text-align: left;
}
.login-form input {
  margin-bottom: 1rem;
}
.log-in {
  margin: 0 auto;
  display: block;
}
