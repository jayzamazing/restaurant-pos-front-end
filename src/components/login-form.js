import React from 'react';
import {Field, reduxForm} from 'redux-form';
import Input from './input';
import {login} from '../actions/auth';
import {required, nonEmpty} from '../validators';
import {connect} from 'react-redux';
import './login-form.css';

export class LoginForm extends React.Component {
    componentWillMount() {
      //deal with setting the initial values for the fields
      this.props.initialize({
        username: 'jack',
        password: 'ifts(8@3re)$'
      });
    }
    render() {
        let error;
        if (this.props.error) {
            error = <div className="form-error">{this.props.error}</div>;
        }
        return (
            <form className="col-centered login-form col-xs-12 col-sm-8 col-md-8 col-md-8 center_div ng-pristine ng-valid" onSubmit={this.props.handleSubmit(values =>
                  this.props.login(values.email, values.password, values.store))}>
                {error}
                <Field
                  label="username"
                  placeholder="Please enter your username"
                  component={Input}
                  type="text"
                  name="username"
                  id="username"
                  validate={[required, nonEmpty]} inputClass="input-field form-control"/>
                <Field
                  label="password"
                  placeholder="Please enter your password"
                  component={Input}
                  type="password"
                  name="password"
                  id="password"
                  validate={[required, nonEmpty]}
                  inputClass="input-field form-control"/>
                  <button className="btn btn-success log-in" disabled={this.props.invalid || this.props.submitting}>Log in</button>
            </form>
        );
    }
}

const mapDispatchToProps = (dispatch) => ({
  //dispatch to authenticate the user
  login: (email, password, store) => {
    dispatch(
      login(email, password, store)
    );
  }
});
export default connect(null, mapDispatchToProps)(reduxForm({form: 'login'})(LoginForm));
