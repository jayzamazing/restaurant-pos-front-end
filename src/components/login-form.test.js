'use strict';
import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import {LoginForm} from './login-form';

describe('login-form', () => {
  //ensure that it renders without issues
  it('should render', () => {
    const handleSubmit = jest.fn();
    shallow(<LoginForm handleSubmit={handleSubmit}/>);
  });
  //take snapshot for future changes
  it('snapshot', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<LoginForm handleSubmit={handleSubmit}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  //click on login button and check dispatches
  it('login button click', () => {
    const handleSubmit = jest.fn();
    const wrapper = shallow(<LoginForm handleSubmit={handleSubmit}/>);
    wrapper.find('[name="email"]').at(0).simulate('change', { target: { value: 'test@test.com' } });
    wrapper.find('[name="password"]').at(0).simulate('change', { target: { value: 'somereallylongpassword2343' } });
    wrapper.find('.log-in').simulate('submit');
    expect(handleSubmit).toHaveBeenCalled();
  });
});
