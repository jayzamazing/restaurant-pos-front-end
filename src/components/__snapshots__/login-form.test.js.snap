// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login-form snapshot 1`] = `
<form
  className="login-form"
>
  <Field
    component={[Function]}
    id="email"
    inputClass="input-field form-control"
    label="Email (or username)"
    name="email"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    id="password"
    inputClass="input-field form-control"
    label="Password"
    name="password"
    type="password"
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <button
    className="btn btn-success log-in"
  >
    Log in
  </button>
</form>
`;
